### ISSUE TITLE
Freeze when importing large XLSX file

### ISSUE THREAD

**Issue by stink0r** [codingchili/excelastic#56]
Hi Robin,

I love your tool, it's wonderful! It only has one problem: If I am trying to import a 100.000 rows XLSX-file with 150 columns, the execution never returns (I was waiting > 15min), even if I give it 8G of memory via the xmx-parameter. The memory is consumed, but nothing happens then. Not even the index is created in elastic after 15min. Splitting the file manually to ~14.000 rows works and imports the data within ~1min – even though some 

`"io.vertx.core.impl.BlockedThreadChecker
WARNING: Thread Thread[vert.x-eventloop-thread-3,5,main] has been blocked for 47
io.vertx.core.VertxException: Thread blocked" `

exceptions are thrown, the data seems to be transferred to elastic completely and without any loss (AFAIK).

Any tips for me how I can manage even to import the large file at once, saving to manually split it?

Thanks!

**Comment by codingchili** [comment]
Hello!

Thanks for submitting an issue.

That sure is one big excel file. The problem here is that the application attempts to parse the full excel file in memory as json, to make sure that it can parse the whole file. Because we'd like to avoid having the import fail halfway due to a parsing error, leaving the index in an undesired state. If it runs out of out of memory in doing so, it will stall indefinitely until more memory is available.

The real solution here would be to save the file to disk and stream it twice, first to parse it and make sure it is valid and then parse it again when importing. When importing we should only parse a few hundred rows at a time.

Additionally I will fix the warning message.

There is a similar issue already in #44, I will add some priority to fixing this and leave both issues open.

**Comment by codingchili** [comment]
Hello again,

I have fixed #44 and made a few changes to performance.

The improvements to memory usage is much smaller than anticipated, as Apache POI is the main consumer of memory here. Anyways, expect imports to be slightly faster in 1.2.7 and memory usage to be improved minimally.

I increased the timeout for the blocked thread checker so these warnings should be less frequent as well.  The size of each uploaded bucket was halved as well.

If you still have issues let me know, its seems to me that POI is the blocker on this for now. Its highly unlikely that I will replace POI with something more efficient. A possible solution could have been to programatically split the file using POI, but in order to do that we still need to load the full file into memory - so no go there either.

Thanks for submitting an issue.
